### PR TITLE
Feat/lib 258 navlist component

### PR DESCRIPTION
**Pull request by Trebh**
Jira:
https://fiscozen.atlassian.net/browse/LIB-258

da mergiare dopo
https://github.com/fiscozen/design_system/pull/7

### CODE REVIEW

File: packages/navlist/src/FzNavlist.vue [hunk 1]
```diff
@@ -0,0 +1,59 @@
+<script lang="ts" setup>
+import { FzRouterNavlinkProps, FzNavlink } from '@fiscozen/navlink'
+import { FzCollapse } from '@fiscozen/collapse'
+import { FzNavlistProps, FzNavlistSub } from './types'
+
+defineProps<FzNavlistProps>()
+
+const isSubMenu = (item: FzRouterNavlinkProps | FzNavlistSub): item is FzNavlistSub => {
+  return Boolean((item as FzNavlistSub).subitems)
```
Comment (oxcened) [inline packages/navlist/src/FzNavlist.vue:9]:
```suggestion
  return "subitems" in item
```
you could as well

File: packages/navlist/src/FzNavlist.vue [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    <div
+      class="fz__navlist__section border-grey-200"
+      v-for="(section, index) in sections"
+      :key="index"
+    >
+      <div class="text-grey-400 flex h-32 items-center px-12 text-xs">
+        <span>{{ section.label }}</span>
+      </div>
+      <div class="flex flex-col" v-for="(item, itemid) in section.items" :key="itemid">
+        <FzCollapse v-if="isSubMenu(item)" :summary-class="'px-12'">
```
Comment (oxcened) [inline packages/navlist/src/FzNavlist.vue:24]:
```suggestion
        <FzCollapse v-if="isSubMenu(item)" summary-class="px-12">
```

File: packages/navlist/src/FzNavlist.vue [hunk 3]
```diff
@@ -0,0 +1,59 @@
+<script lang="ts" setup>
+import { FzRouterNavlinkProps, FzNavlink } from '@fiscozen/navlink'
+import { FzCollapse } from '@fiscozen/collapse'
+import { FzNavlistProps, FzNavlistSub } from './types'
+
+defineProps<FzNavlistProps>()
+
+const isSubMenu = (item: FzRouterNavlinkProps | FzNavlistSub): item is FzNavlistSub => {
```
Comment (oxcened) [inline packages/navlist/src/FzNavlist.vue:8]: I think this type guard belongs in `types.ts`

File: packages/navlist/src/FzNavlist.vue [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  <div class="fz__navlist inline-flex grow-0 flex-col rounded p-4">
+    <div
+      class="fz__navlist__section border-grey-200"
+      v-for="(section, index) in sections"
+      :key="index"
+    >
+      <div class="text-grey-400 flex h-32 items-center px-12 text-xs">
+        <span>{{ section.label }}</span>
+      </div>
+      <div class="flex flex-col" v-for="(item, itemid) in section.items" :key="itemid">
```
Comment (oxcened) [inline packages/navlist/src/FzNavlist.vue:23]: itemid is not actually an id, a bit misleading

**Review by oxcened — COMMENTED** [review]
some small changes but looks good overall

File: packages/navlist/src/FzNavlist.vue (hunk 3 above)
Comment (Trebh) [reply packages/navlist/src/FzNavlist.vue:8]: fair, usually I tend not to have runtime definitions in "types" files but this is strictly related to typings

File: packages/navlist/src/FzNavlist.vue (hunk 4 above)
Comment (Trebh) [reply packages/navlist/src/FzNavlist.vue:23]: I concur. I also renamed the "index"es in the file to at least have some sort of identifier (there were 3 index for sections, items and subitems)

**Review by oxcened — APPROVED** [review]

**Pull request merged**